nexus/src/lib.rs

```diff
@@ -417,6 +418,42 @@ pub async fn run_server(config: &NexusConfig) -> Result<(), String> {
     }
     let internal_server = InternalServer::start(config, &log).await?;
     let server = Server::start(internal_server).await?;
-    server.register_as_producer().await;
     server.wait_for_finish().await
 }
+
+/// Create a new metric producer server.
+fn start_producer_server(
+    log: &Logger,
+    registry: &ProducerRegistry,
+    nexus_addr: SocketAddr,
+) -> Result<ProducerServer, String> {
+    // The producer server should listen on any available port, using the
+    // same IP as the main Dropshot server.
+    let address = SocketAddr::new(nexus_addr.ip(), 0);
+
+    // Create configuration for the server.
+    //
+    // Note that because we're registering with _ourselves_, the listening
+    // address for the producer server and the registration address use the
+    // same IP.
+    let config = oximeter_producer::Config {
+        server_info: ProducerEndpoint {
+            id: registry.producer_id(),
+            kind: ProducerKind::Service,
+            address,
+            base_route: String::from("/metrics/collect"),
```

Comment:
Just double-checking - `base_route` is irrelevant now right?

Reply:
Yeah, it jumped out at me because the rest of the ones in the diff have become an empty string with a comment about it being unused.

Reply:
Cool, I'll change this one too, thanks for catching.